pkg/apply/destroyer.go

```diff
@@ -71,6 +72,43 @@ func setDestroyerDefaults(o *DestroyerOptions) {
 	}
 }
 
+// prepareObjects returns the set of objects to prune or an error if one occurred.
+func (d *Destroyer) prepareObjects(
```

Comment:
I wonder if this refactor, to move the Pruner.GetPruneObjs code into Destoryer.prepareObjects could be done by itself in a separate PR. It would have to go first, and it wouldn't look exactly like this, because it would use the old inventory info and client, but it would at least move the code into the right place, which would make the diff easier to follow in this PR.

Does that sound worth the extra effort?